src/Tribe/Ignored_Events.php

```diff
@@ -0,0 +1,526 @@
+
+			if ( empty( $post_type->cap->edit_posts ) || ! current_user_can( $post_type->cap->edit_posts ) ) {
+				$response->status = false;
+				$response->text = esc_html__( 'You do not have permission to migrate Legacy Ignored Events', 'the-events-calendar' );
+
+				wp_send_json( $response );
+			}
+
+			if ( ! $this->has_legacy_deleted_posts() ) {
+				$response->status = true;
+				$response->text = esc_html__( 'There were no Legacy Events to be Migrated, you are ready to rock!', 'the-events-calendar' );
+
+				wp_send_json( $response );
+			}
+
+			$args = array(
+				'post_type' => self::$legacy_deleted_post,
+				'post_status' => 'trash',
+			);
+			$query = new WP_Query( $args );
+
+			foreach ( $query->posts as $event ) {
+				$status = $this->maybe_convert_legacy_post( $event );
+				if ( is_wp_error( $status ) ) {
+					$response->error[ $event->ID ] = $status->get_error_message();
+				} else {
+					$response->migrated[ $event->ID ] = true;
+				}
+			}
+
+			if ( ! empty( $response->error ) ) {
+				if ( ! empty( $response->migrated ) ) {
+					$response->status = false;
+					$response->text = sprintf(
+						_n(
+							'Migration: %d Legacy Ignored Post was migrated but %d failed.',
+							'Migration: %d Legacy Ignored Posts were migrated but %d failed.',
+							count( $response->migrated ),
+							'the-events-calendar'
+						),
+						count( $response->migrated ),
+						count( $response->error )
+					);
+				} else {
+					$response->status = false;
+					$response->text = sprintf(
+						_n(
+							'Migration: %d Legacy Ignored Post failed to be migrated.',
+							'Migration: %d Legacy Ignored Posts failed to be migrated.',
+							count( $response->error ),
+							'the-events-calendar'
+						),
+						count( $response->error )
+					);
+				}
+
+				$response->text .= '<ul>';
+				foreach ( $response->error as $ID => $message ) {
+					$response->text .= '<li>' . sprintf( __( 'Event %d: %s', 'the-events-calendar' ), $ID, $message ) . '</li>';
+				}
+				$response->text .= '</ul>';
+
+			} elseif ( ! empty( $response->migrated ) ) {
+				$response->status = true;
+				$response->text = sprintf(
+					_n(
+						'Migration: %d Legacy Ignored Post was migrated sucessfully.',
+						'Migration: %d Legacy Ignored Posts were migrated sucessfully.',
+						count( $response->migrated ),
+						'the-events-calendar'
+					),
+					count( $response->migrated )
+				);
+			}
+
+			wp_send_json( $response );
+		}
+
+		/**
+		 * Notes:
+		 *
+		 * Working on this I notice that because `pre_delete_post` only exists after WP 4.4 we have a problem in terms of preventing that post from been deleted.
+		 * My workout for this is to hoot on `admin_init` of that page, change the `$_REQUEST` variable holding the IDs to remove any posts that fit the `ignored`
+		 *
+		 * On another problem, I realized that not WordPress still not ready to display custom status on the Admin, how should I display if we have a custom status
```

Comment:
Could be a good discussion for team CR on Wednesday - I've got a related ticket where ideally we'd also be able to render custom statuses.